{"cells":[{"cell_type":"markdown","metadata":{},"source":["## use sourmash to make a depth file similar to read depth but based on kmers\n","- symlink the read sketches to folder\n","- create a file from all contigs\n","- sketch the contigs into kmers (k21,31,51)\n","- collect the sigs into sql db\n","- Run fastmultigather of all contigs <-> all reads\n","- Run a mgmanysearch using fmg result as picklist, using snakemake\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# srun \n","srun --account=ctbrowngrp -p bmm -J sketch -t 5:00:00 -c 1 --mem 10gb --pty bash"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# symlink the reads sigs\n","ln -s ../../../2023-swine-sra/sourmash/sig_files/sketch_reads/ERR113518* .\n","ln -s ../../../2023-swine-sra/sourmash/sig_files/sketch_reads/ERR113517* ."]},{"cell_type":"markdown","metadata":{},"source":["# Use manysearch to look for abundance of contigs in metag\n","- Does not work with full read files right now\n","- Try the following:\n","    - Run manysearch with a subsetted read file, made from forward and reverse reads\n","    - then try with the full read file\n","    \n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":[]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# get manysearch to work...\n","# first on small subset of reads (WORKS)\n","# make small set (get the first 100 reads)\n","gunzip -c /group/ctbrowngrp2/scratch/annie/2023-swine-sra/atlas/atlas_ERR1135178/ERR1135178/sequence_quality_control/ERR1135178_QC_R2.fastq.gz | head -n 1000 | gzip > ERR1135178_QC_R2.subset.fq.gz\n","gunzip -c /group/ctbrowngrp2/scratch/annie/2023-swine-sra/atlas/atlas_ERR1135178/ERR1135178/sequence_quality_control/ERR1135178_QC_R1.fastq.gz | head -n 1000 | gzip > ERR1135178_QC_R1.subset.fq.gz\n","\n","# create a sig file\n","sourmash sketch dna \\\n","-p k=21,k=31,k=51,abund,scaled=100 \\\n","ERR1135178_QC_R1.subset.fq.gz ERR1135178_QC_R2.subset.fq.gz \\\n","--name ERR1135178 -o ../sketch_reads/ERR1135178_QC.subset.zip\n","\n","# manysearch\n","sourmash scripts manysearch \\\n","../sourmash_sketches/sketch_reads/ERR1135178_QC.subset.zip \\\n","../sourmash_sketches/ERR11351.zip \\\n","-k 21 --scaled 100 -o ERR1135178_QC.subset.csv -c 1\n","\n","#Now full read set, sketch and manysearch the exact same way:\n","sourmash sketch dna \\\n","-p k=21,k=31,k=51,abund,scaled=100 \\\n","ERR1135178_QC_R1.fastq.gz ERR1135178_QC_R2.fastq.gz \\\n","--name ERR1135178 -o ../sketch_reads/ERR1135178.zip\n","\n","sourmash scripts manysearch \\\n","../sourmash_sketches/sketch_reads/ERR1135178.zip \\\n","../sourmash_sketches/ERR11351.zip \\\n","-k 21 --scaled 100 -o ERR1135178.csv -c 1\n","\n"]},{"cell_type":"markdown","metadata":{},"source":["## Compare mgmanysearch and manysearch output\n","- Create a small subset of data\n","- Run manysearch and mgmanysearch on these\n","- Get manysearch to run in general...\n","\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# make small set (get the first 100 reads)\n","gunzip -c /group/ctbrowngrp2/scratch/annie/2023-swine-sra/atlas/atlas_ERR1135178/ERR1135178/sequence_quality_control/ERR1135178_QC_R2.fastq.gz | head -n 100 | gzip > ERR1135178_QC_R2.subset.fq.gz\n","\n","# create a sig file\n","sourmash sketch dna \\\n","-p k=21,k=31,k=51,abund,scaled=100 \\\n","ERR1135178_QC_R2.subset.fq.gz \\\n","--name ERR1135178 -o ../sketch_reads/ERR1135178_QC_R2.subset.zip\n","\n","# run manysearch\n","sourmash scripts manysearch \\\n","../sourmash_sketches/sketch_reads/ERR1135178_QC_R2.subset.zip \\\n","../sourmash_sketches/ERR11351.zip \\\n","-k 21 --scaled 100 -o ERR1135178_QC_R2.subset.csv -c 1\n","\n","# grep the successful matches so we have to compare less\n","sourmash sig grep 'ERR1135178_5527' ERR11351.zip -o ERR11351_5527.zip\n","\n","# and mgmanysearch: This works, but the output file created has abundances of 1 \n","sourmash scripts mgmanysearch \\\n","--against ../sourmash_sketches/sketch_reads/ERR1135178_QC_R2.subset.zip \\\n","--queries ../sourmash_sketches/ERR11351_5527.zip \\\n","-k 21 --scaled 100 -o ERR1135178_QC_R2.subset.mgm.csv "]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":[]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# retry at a scaled of 100, treshold of 10\n","# sketch the reads, save as sig.gz \n","# make a zip for each fasta, concat those then make a mf\n","for f in *_R1.fastq.gz\n","do\n","echo sourmash sketch dna \\\n","-p abund,k=31,scaled=100,abund,k=21,scaled=100,abund,k=51,scaled=100 $f ${f%_R1*}_R2.fastq.gz \\\n","--name ${f%_QC*} -o ${f%_QC*}.abund.sig.gz \n","done | parallel -j 40\n","\n","\n","sourmash sketch dna \\\n","-p abund,k=31,scaled=100,abund,k=21,scaled=100,abund,k=51,scaled=100 \\\n","ERR1135186_QC_R1.fastq.gz \\\n","--name ERR1135186 -o ERR1135186_R1.abund.zip\n","done | parallel -j 16\n","\n","\n","\n","#manysketch??"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# run the snakefile\n","\n","srun --account=ctbrowngrp -p med2 -J sketch -t 8:30:00 -c 4 --mem=50gb --pty bash\n","mamba activate branchwater\n","snakemake --resources mem_mb=50000 --rerun-triggers mtime \\\n","-c 24 --rerun-incomplete -k --latency-wait 1 -n\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# try manysearch\n","srun --account=ctbrowngrp -p high2 -J bwabund -t 1:00:00 -c 100 --mem=50gb --pty bash\n","mamba activate branchwater-abund\n","\n","\n","# manysearch\n","mamba activate branchwater-abund\n","sourmash scripts manysearch \\\n","../sourmash_sketches/sketch_reads/ERR1135186.abund.sig.gz \\\n","../sourmash_sketches/ERR11351.zip \\\n","-k 21 --scaled 100 -o ERR1135186.fullread.fulldb.csv -c 100\n","\n","# # normal mgmany\n","# mamba activate branchwater\n","# sourmash scripts mgmanysearch \\\n","# --against ../sourmash_sketches/sketch_reads/ERR1135181.k21.sig.gz \\\n","# --queries ../sourmash_sketches/split_sig/ERR11351_01.zip \\\n","# -k 21 --scaled 100 -o fmg_mgmanysearch.csv -c 16\n","\n","sourmash sketch dna \\\n","-p abund,k=31,scaled=100,abund,k=21,scaled=100,abund,k=51,scaled=100 \\\n","ERR1135186_QC_R1.fastq.gz \\\n","-o ERR1135186_R1.abund.noname.zip\n","\n","\n","sourmash scripts manysearch \\\n","../sourmash_sketches/sketch_reads/ERR1135186_R1.abund.zip \\\n","../test_mgsearch/ERR11351_3contig.zip \\\n","-k 21 --scaled 100 -o ERR1135186_testwfullreads_smalldb.csv -c 2\n","\n","sourmash scripts manysearch \\\n","../test_mgsearch/ERR11351_R1.zip \\\n","../sourmash_sketches/ERR11351.zip \\\n","-k 21 --scaled 100 -o ERR11351_testwsmallreads_fulldb.csv -c 100\n","\n","\n","\n","sourmash scripts manysearch \\\n","../sourmash_sketches/sketch_reads/ERR1135186_R1.k21.abund.zip \\\n","../test_mgsearch/ERR11351_3contig.zip \\\n","-k 21 --scaled 100 -o ERR1135186_testwfullreads_smalldb.csv -c 2"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["sourmash scripts manysearch \\\n","../sourmash_sketches/sketch_reads/all_reads.zip \\\n","ERR11351_3contig.zip \\\n","-k 21 --scaled 100 -o all_reads_smalldb.csv -c 10"]}],"metadata":{"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":2}
