{"cells":[{"cell_type":"markdown","metadata":{},"source":["## use sourmash to make a depth file similar to read depth but based on kmers\n","- symlink the read sketches to folder\n","- create a file from all contigs\n","- sketch the contigs into kmers (k21,31,51)\n","- collect the sigs into sql db\n","- Run fastmultigather of all contigs <-> all reads\n","- Run a mgmanysearch using fmg result as picklist, using snakemake\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# srun \n","srun --account=ctbrowngrp -p bmm -J sketch -t 10:00:00 -c 4 --mem 10gb --pty bash"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# symlink the reads sigs\n","ln -s ../../../2023-swine-sra/sourmash/sig_files/sketch_reads/ERR113518* .\n","ln -s ../../../2023-swine-sra/sourmash/sig_files/sketch_reads/ERR113517* ."]},{"cell_type":"markdown","metadata":{},"source":["## Run manysearch to score abundance of contig in each metagenome\n","- manysearch needs a treshold of 0, because there will never be 1% of the reads linked to the genome, bc its a shit ton of reads. \n","- Compare manysearch to mgmanysearch\n","- Run Tessa's scripts to create a depth file from manysearch output\n","\n","### Compare mgmanysearch and manysearch output\n","- Create a small subset of data\n","- Run manysearch and mgmanysearch on these\n","\n","\n","Workflow:\n","- sketch all reads with manysketch (abundance, k21,k31,k51)\n","- sketch the contigs (no abundance, k21,k31,k51)\n","- run manysearch (k31, scaled=100, t=0)\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# manysketch\n","# How to run manysketch??\n","# https://github.com/sourmash-bio/sourmash_plugin_branchwater/tree/main/doc#running-manysketch\n","# one file for all the sketches?\n","\n","# make a read file\n","echo name,read1,read2 > manysketch.csv\n","for i in *_R1.fastq.gz\n","do\n","echo ${i%_QC*},$i,${i%_R1*}_R2.fastq.gz\n","done >> manysketch.csv\n","\n","#sketch from readfile\n","mamba activate branchwater\n","sourmash scripts manysketch manysketch.csv \\\n","-p k=21,k=31,k=51,scaled=100,abund -c 100 \\\n","-o ../all_reads.resketch.zip"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# sketch contigs\n","# symlink the atlas folders\n","ln -s /group/ctbrowngrp2/scratch/annie/2023-swine-sra/atlas/atlas_ERR113517*/\n","ln -s /group/ctbrowngrp2/scratch/annie/2023-swine-sra/atlas/atlas_ERR113518*/\n","\n","# cat all contigs to a results file (n=349,706)\n","cat atlas_*/*/*_contigs.fasta > ../../results/sourmash_sketches/ERR11351.fasta\n","\n","# mamba activate branchwater\n","sourmash sketch dna -p k=21,scaled=100,k=31,scaled=100,k=51,scaled=100 \\\n","--singleton ERR11351.fasta -o ERR11351.zip"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# run manysearch (from ./results/manysearch/)\n","mamba activate branchwater-abund\n","sourmash scripts manysearch \\\n","../sourmash_sketches/all_reads.zip \\\n","../sourmash_sketches/ERR11351.zip \\\n","-k 31 --scaled 100 -o allreads_x_allcontigs.k31.csv -c 100 -t 0"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# create a contig length file and the depth file for binning\n","python ../workflow/scripts/get-contig-lengths.py ../resources/ERR11351.fasta depth_files/contig_lengths.csv\n","\n","# make a list of input csv files (in this case just 1)\n","readlink -f allreads_x_allcontigs.k31.csv > file_list.txt\n","\n","# create the depth file\n","python ../workflow/scripts/manysearch-to-mb-depth.py \\\n","./manysearch/file_list.txt \\\n","--lengths ./depth_files/contig_lengths.csv \\\n","-o ./depth_files/depth_manysearch.txt"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# Now the goal is to use this depth file in metabat and or vamb and does it improve bins?\n","# creating bins with either vamb or metabat:\n","\n","# run with intravariance var\n","# METABAT\n","srun --account=ctbrowngrp -p med2 -J metabat2 -t 10:00:00 -c 8 --mem=40gb --pty bash\n","mamba activate metabat2\n","metabat2 -m 1500 \\\n","-i ./resources/ERR11351.fasta \\\n","-o ./results/manysearch/metabat \\\n","-a ./results/depth_files/depth_manysearch.txt -t 8\n"," \n"," # VAMB\n","srun --account=ctbrowngrp -p med2 -J vamb -t 10:00:00 -c 1 --mem=40gb --pty bash\n","mamba activate vamb\n","vamb --outdir ./results/manysearch/vamb \\\n","--fasta ./resources/ERR11351.fasta \\\n","--jgi ./results/depth_files/depth_manysearch.txt \\\n","--minfasta 50000\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":[]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# compare manysearch and mgmanysearch\n","# subset a set of reads (n=100,000):\n","gunzip -c /group/ctbrowngrp2/scratch/annie/2023-swine-sra/atlas/atlas_ERR1135178/ERR1135178/sequence_quality_control/ERR1135178_QC_R1.fastq.gz | head -n 100000 | gzip > ERR1135178_QC_R1.subset.fq.gz\n","gunzip -c /group/ctbrowngrp2/scratch/annie/2023-swine-sra/atlas/atlas_ERR1135178/ERR1135178/sequence_quality_control/ERR1135178_QC_R2.fastq.gz | head -n 100000 | gzip > ERR1135178_QC_R2.subset.fq.gz\n","\n","# create a signature\n","mamba activate branchwater\n","sourmash sketch dna \\\n","-p k=21,k=31,k=51,abund,scaled=100 \\\n","ERR1135178_QC_R1.subset.fq.gz ERR1135178_QC_R2.subset.fq.gz \\\n","--name ERR1135178 -o ../sketch_reads/ERR1135178_QC.subset.zip\n","\n","# subset the fasta file for 1000 contigs\n","mamba activate bbmap\n","reformat.sh in=ERR11351.fasta out=ERR11351_1000.fasta reads=1000\n","\n","# sketch\n","mamba activate branchwater\n","sourmash sketch dna -p k=21,scaled=100,k=31,scaled=100,k=51,scaled=100 \\\n","--singleton ERR11351_1000.fasta -o ERR11351_1000.zip\n","\n","# mgmanysearch for spot check (from ./results/mgmanysearch) and timed\n","mamba activate branchwater\n","/usr/bin/time -v sourmash scripts mgmanysearch \\\n","--against ../sourmash_sketches/sketch_reads/ERR1135178_QC.subset.zip \\\n","--queries ../sourmash_sketches/ERR11351_1000.zip \\\n","-k 31 --scaled 100 -o ERR1135178.timed.mgm.csv\n","\n","# manysearch\n","mamba activate branchwater-abund\n","/usr/bin/time -v sourmash scripts manysearch \\\n","../sourmash_sketches/sketch_reads/ERR1135178_QC.subset.zip \\\n","../sourmash_sketches/ERR11351_1000.zip \\\n","-k 31 --scaled 100 -o ERR1135178.mn.timed.csv -c 1 -t 0"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# run a snakefile\n","srun --account=ctbrowngrp -p med2 -J sketch -t 8:30:00 -c 4 --mem=50gb --pty bash\n","mamba activate branchwater\n","snakemake --resources mem_mb=50000 --rerun-triggers mtime \\\n","-c 24 --rerun-incomplete -k --latency-wait 1 -n\n"]}],"metadata":{"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":2}
