{"cells":[{"cell_type":"markdown","metadata":{},"source":["## use sourmash to make a depth file similar to read depth but based on kmers\n","- symlink the read sketches to folder\n","- create a file from all contigs\n","- sketch the contigs into kmers (k21,31,51)\n","- collect the sigs into sql db\n","- Run fastmultigather of all contigs <-> all reads\n","- Run a mgmanysearch using fmg result as picklist, using snakemake\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# symlink the reads sigs\n","ln -s ../../../2023-swine-sra/sourmash/sig_files/sketch_reads/ERR113518* .\n","ln -s ../../../2023-swine-sra/sourmash/sig_files/sketch_reads/ERR113517* ."]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# srun \n","srun --account=ctbrowngrp -p bmm -J fmg_bin -t 24:00:00 -c 100 --mem 100gb --pty bash"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# retry at a scaled of 100, treshold of 10\n","# sketch the reads, save as sig.gz \n","# make a zip for each fasta, concat those then make a mf\n","for f in *_R1.fastq.gz\n","do\n","echo sourmash sketch dna \\\n","-p k=31,scaled=100 $f ${f%_R1*}_R2.fastq.gz \\\n","--name ${f%_QC*} -o ${f%_QC*}.k31.sig.gz \n","done | parallel -j 24"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# fastmultigather with scale of 100.\n","\n","mamba activate branchwater\n","sourmash scripts fastmultigather \\\n","../sketch_reads/.k21.txt \\\n","../sketch_contigs/ERR11351.k21.zip \\\n","-c 100 -k 21 -t 1000 -s 100\n","\n","mamba activate branchwater\n","sourmash scripts fastmultigather \\\n","../sketch_reads/ERR1135178.k21.zip \\\n","../sketch_contigs/ERR11351.k21.zip \\\n","-c 100 -k 21 -t 1000 -s 100"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# can give it a list of query files!!\n","sourmash scripts mgmanysearch \\\n","--queries sketch_contigs/ERR1135178.sig \\\n","--against sketch_reads/ERR1135178.sig.gz \\\n","-k 21 --scaled 1000 -o ERR1135178.mgm.csv"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# run the snakefile\n","\n","srun --account=ctbrowngrp -p med2 -J sketch -t 8:30:00 -c 4 --mem=50gb --pty bash\n","mamba activate branchwater\n","snakemake --resources mem_mb=50000 --rerun-triggers mtime \\\n","-c 4 --rerun-incomplete -k --latency-wait 1 -n\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# do you remove sigs with n=1?\n","# these aren't abundance sketched so wont work\n","\n","# split the fasta\n","# use bbmap (gives us Â±3500 per file)\n","mamba activate bbmap\n","partition.sh in=../ERR11351.fasta out=./ERR11351_%.fa ways=100\n","\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# split the fasta contig file into many smaller dbs\n","\n","# then fmg for each smaller db per read file\n","sourmash fastmultigather <metagenome> <database-split-N> -o <database-split-N.csv>\n","\n","# fastmultigather --output only works with a rocksdb..\n","\n","\n","# concat the picklists\n","concatenate the resulting <database-split-N.csv> files with csvtk concat\n","\n","# run mgmanysearch using this combined picklist to all the db files. \n","# create picklist first\n","sourmash sig check -k 21 <database-split-*.zip> --picklist <combined_csv>:match_md5:md5 -m <manifest>\n","\n","# mgmany usig a manifest\n","sourmash scripts mgmanysearch --queries <manifest> \\\n","--against <reads> -k 21 --scaled 100 -o {output.csv} "]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["# create a rocksdb and run it?\n","# needs to be done from a sig.gz file\n","sourmash sketch dna -p k=21,scaled=100 --singleton ERR11351_0.fa -o ../split_sig/ERR11351_0.sig.gz\n","\n","# make a txt file pointing to the sig.gz\n","readlink -f  ERR11351_0.zip > ERR11351_0.txt\n","\n","# create rocksdb\n","sourmash scripts index \\\n","ERR11351_0.txt -m DNA -k 21 --scaled 100 \\\n","-o ../rocksdb/ERR11351_0.rocksdb\n","\n","\n","sourmash scripts fastmultigather \\\n","../sketch_reads/ERR1135178.k21.zip \\\n","ERR11351_0.rocksdb -c 4 -k 21 -t 1000 -s 100 -m DNA \\\n","-o ../results/sourmash/fastmultigather/ERR1135178xERR11351_0.csv"]}],"metadata":{"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":2}
